### DISCUSSION TITLE
Determine that there are human signatures on my pdf file

### DISCUSSION THREAD

**Discussion by paulgromy** [mindee/doctr#1272] [Q&A] [answered]
I recognize pdf files and would like to understand how to determine that a pdf file has a person's signature. According to the documentation, I should use Artefact. But I can't figure out how to apply it in code. If I take the standard code from the documentation, then in the output Document all the artifacts are an empty list (although there are human signatures on the pdf file).

Tell me how can I determine that there are human signatures on my pdf file

**Comment by felixT2K** [comment, marked as answer]
Hi @paulgromy 👋🏼 ,

the artefact detection supports currently the following classes: 

`CLASSES = ["__background__", "QR Code", "Barcode", "Logo", "Photo"]`

In docTR we don't have currently any logic to determine human signatures you could try to solve this with training your own KIE model if you have a labeled dataset for training and evaluation :)

**Reply by paulgromy** [reply]
Thank you very much for your reply. You may also be able to suggest about the handwritten text. I just didn't find any implementation examples in the documentation. Let's say I have a PDF file that contains handwritten text in addition to printed text. Can I recognize it with docTR?

**Reply by felixT2K** [reply]
The available models in docTR are only trained on printed text so if you want to recognize handwritten you need to fine tune a model on your own data. But it's on the list of tasks we will work on. :)
